storagenode/storagenodedb/database.go

```diff
@@ -41,29 +43,10 @@ var (
 
 var _ storagenode.DB = (*DB)(nil)
 
-// SQLDB defines interface that matches *sql.DB
-// this is such that we can use utccheck.DB for the backend
-//
-// TODO: wrap the connector instead of *sql.DB
+// SQLDB defines an interface to allow accessing and setting an sql.DB
 type SQLDB interface {
-	Begin() (*sql.Tx, error)
-	BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
-	Close() error
-	Conn(ctx context.Context) (*sql.Conn, error)
-	Driver() driver.Driver
-	Exec(query string, args ...interface{}) (sql.Result, error)
-	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
-	Ping() error
-	PingContext(ctx context.Context) error
-	Prepare(query string) (*sql.Stmt, error)
-	PrepareContext(ctx context.Context, query string) (*sql.Stmt, error)
-	Query(query string, args ...interface{}) (*sql.Rows, error)
-	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
-	QueryRow(query string, args ...interface{}) *sql.Row
-	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
-	SetConnMaxLifetime(d time.Duration)
-	SetMaxIdleConns(n int)
-	SetMaxOpenConns(n int)
+	Configure(sqlDB *sql.DB)
+	GetDB() *sql.DB
```

Comment:
When you have a getter there's no need to prefix it with `Get`, usually at least. You can read about it in here https://golang.org/doc/effective_go.html#Getters.

Reply:
Yes, but here the name `DB` conflicts with the embedded `*sql.DB`, which is why we had to come up with something else. I knew this one would draw a comment. :smile: I'm fine picking something else besides `GetDB()`, but something more explicit like `SQLDB()` stuttres.

Reply:
Ah, yeah, in that case `GetDB` is probably the next best thing. However I'm wondering whether there's a way to get rid of `SQLDB` altogether.

Reply:
We played with that as well, let me know if you find something that seems better. We need some way to swap out the underlying sql.DB object for the various db objects.

Reply:
Where do you need to swap out the sql.DB? Do you mean configure, or are there other places.

Otherwise one approach would be to replace SQLDB with migratableDB everywhere... of course adjusting things to make it nicer.

Reply:
Yeah we swap it out in `Configure()`. Can I create a separate ticket to address this clean up? It doesn't seem like we should hold testing this feature back because we want to refactor and remove `SQLDB`.